src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +665,36 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private IReadOnlyList<PackageSource> GetAllPackageSources()
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => { return await GetAllPackageSourcesAsync(); });
```

Comment:
You must not use JTF.Run around access of brokered services. These UI delays can be *extreme*, and you can deadlock when/if the server can't satisfy the request without a call back to the client when you block the UI thread.
Instead, please refactor to display the UI asynchronously.

Reply:
I was looking to see what DialogPage did as these methods are ultimately called from there and they are also wrapping JTF calls around brokered service access.

Is there an example of someone doing this right? Since this is happening during UI initialization of our options page, should we be displaying a loading progress? What are the guidelines?